### PR TITLE
LPS-164161 Add the expiration date and review date to database and LocalService method changes

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
This is the **First** PR to add the new fields to Knowledge Base Articles, This adds the 2 new fields: expirationDate and reviewDate to the KBArticle Table and changes the methods for the LocalService.

Enjoy!

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:stop

**Comment by AliciaGarciaGarcia** [comment]
I talked with upgrade team and now there is a new way to make the add columns, i will add the new way on a commit

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

File: modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/internal/upgrade/v4_3_0/UpgradeKBArticle.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		if (!hasColumn("KBArticle", "expirationDate")) {
+			alterTableAddColumn("KBArticle", "expirationDate", "DATE null");
+		}
+
+		if (!hasColumn("KBArticle", "reviewDate")) {
+			alterTableAddColumn("KBArticle", "reviewDate", "DATE null");
+		}
```
Comment (adolfopa) [inline modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/internal/upgrade/v4_3_0/UpgradeKBArticle.java]: Not necessary for this pr to be forwarded, but it is no longer necessary to check if the column exists. See https://liferay.atlassian.net/wiki/spaces/ENGCOREUPGRADE/blog/2022/08/25/2108196950/Simplify+DB+modification+framework+Part+2

**Review by adolfopa — APPROVED** [review]

File: modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/internal/upgrade/v4_3_0/UpgradeKBArticle.java (hunk 1 above)
Comment (AliciaGarciaGarcia) [reply modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/internal/upgrade/v4_3_0/UpgradeKBArticle.java]: yeah, I realized after that and made an additional commit on [04c6c6a](https://github.com/liferay-lima/liferay-portal/pull/3550/commits/04c6c6a2c5c151e4df13ef8e1cdaa65bef365ead) deleting this class

**Review by marcogalluzzi — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Pull request closed without merging**